### PR TITLE
DSND-3113: Changes for psc delete delta order processing (strict)

### PR DESCRIPTION
**Pull request by crowlandsCH**
* addition of delta_at and kind to the headers sent across to the data-api. 
* Checkstyle plugin removed as unnecessary. 
* All test cases updated and fixed.
* New KindMapper added to replicate upsert behaviour on the delete deltas.

Locally Tested psc upsert and deletion with stream events along with the relevant sdk PRs.

[DSND-3113](https://companieshouse.atlassian.net/browse/DSND-3113)

[DSND-3113]: https://companieshouse.atlassian.net/browse/DSND-3113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/psc/delta/mapper/KindMapper.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package uk.gov.companieshouse.psc.delta.mapper;
+
+import consumer.exception.NonRetryableErrorException;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.delta.PscDeleteDelta.KindEnum;
+
+@Component
+public class KindMapper {
+
+    private KindMapper() {
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/psc/delta/mapper/KindMapper.java:10]: As a component this should not have a private constructor and its method should not be static. We should inject the KindMapper into any classes which depend on it so we can mock its behaviour in a test.

File: src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java [hunk 2]
```diff
@@ -79,7 +80,7 @@ public void processDelta(Message<ChsDelta> chsDelta) {
      */
     public void processDelete(Message<ChsDelta> chsDelta) {
         final ChsDelta payload = chsDelta.getPayload();
-        final String logContext = payload.getContextId();
+        final String contextId = payload.getContextId();
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java:83]: Can `notificationId` be declared on the same line as instantiation?

File: src/main/java/uk/gov/companieshouse/psc/delta/mapper/KindMapper.java (hunk 1 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/psc/delta/mapper/KindMapper.java:10]: Done, comment added to explain why it has an empty constructor but it is a component

File: src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java (hunk 2 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java:83]: done

**Review by cgregoryCH — APPROVED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Pull request merged**